crates/common/tedge_config/src/system_services/log_config.rs

```diff
@@ -28,22 +34,44 @@ pub fn log_init(
         .log_level
         .or(flags.debug.then_some(tracing::Level::DEBUG));
 
-    if let Some(log_level) = log_level {
-        subscriber.with_max_level(log_level).init();
-        return Ok(());
-    }
-
-    if std::env::var("RUST_LOG").is_ok() {
-        subscriber
-            .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+    let log_layer = if let Some(log_level) = log_level {
+        log_layer
+            .with_filter(LevelFilter::from_level(log_level))
+            .boxed()
+    } else if std::env::var("RUST_LOG").is_ok() {
+        log_layer
             .with_file(true)
             .with_line_number(true)
-            .init();
-        return Ok(());
-    }
-
-    let log_level = get_log_level(sname, config_dir)?;
-    subscriber.with_max_level(log_level).init();
+            .with_filter(tracing_subscriber::EnvFilter::from_default_env())
+            .boxed()
+    } else {
+        let log_level = get_log_level(sname, config_dir)?;
+        log_layer
+            .with_filter(LevelFilter::from_level(log_level))
+            .boxed()
+    };
+
+    // Audit journal
+    let audit_appender = tracing_appender::rolling::hourly("/tmp", "tedge.audit.log");
+    let audit_layer = tracing_subscriber::fmt::layer()
+        .with_writer(audit_appender)
+        .with_timer(tracing_subscriber::fmt::time::UtcTime::rfc_3339())
+        .with_filter(LevelFilter::INFO)
+        .with_filter(filter_fn(|metadata| metadata.target() == "Audit"));
+
+    // Actor traces
+    let trace_appender = tracing_appender::rolling::hourly("/tmp", "tedge.actors.log");
```

Comment:
This will have to be configurable ;-)

Reply:
The idea is observe & debug actors and ideally to test actors.

- [ ] Distinguish source messages (MQTT / inotify / service start / ^C) from reactions
- [ ] Add trace identifiers to follow the consequence of an input messages
- [ ] Do we have to interleave actions on the system here (HTTP requests? File Updates? System commands?)

Reply:
> Distinguish source messages (MQTT / inotify / service start / ^C) from reactions
I think ideally we should distinguish between what are strictly inputs/outputs from internal actor messages. To that effect, we do already have `MQTT_sub`, `MQTT_pub` and `MQTT_recv` targets